### ISSUE TITLE
[ImageMask] - Brush missing

### ISSUE THREAD

**Issue by elismasilva** [gradio-app/gradio#10236]
### Describe the bug

I don't know what I'm doing wrong, or if it's a bug, but I'm trying to use the brushes to draw the mask as is done in this example:
https://huggingface.co/spaces/OzzyGT/diffusers-image-fill

But when I load exactly the same code it doesn't work.
In this docs https://www.gradio.app/docs/gradio/imageeditor#demos i tried upload image here but has same problem.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr
with gr.Blocks() as app:   
    imageMask = gr.ImageMask(
            type="pil",
            label="Input Image",
            crop_size=(1024, 1024),
            canvas_size=(1024, 1024),
            layers=False,
            sources=["upload"],
            interactive=True,
            width=400,
            height=400,

        )

app.launch(inbrowser=True)
```

### Screenshot

![image](https://github.com/user-attachments/assets/e1b982af-502f-4786-8aea-c914a962f5fd)

### Logs

```shell
N/A
```

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Windows
gradio version: 5.9.1
gradio_client version: 1.5.2

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
anyio: 4.4.0
audioop-lts is not installed.
fastapi: 0.115.4
ffmpy: 0.4.0
gradio-client==1.5.2 is not installed.
httpx: 0.27.0
huggingface-hub: 0.25.2
jinja2: 3.1.3
markupsafe: 2.1.5
numpy: 1.26.3
orjson: 3.10.6
packaging: 24.1
pandas: 2.2.2
pillow: 11.0.0
pydantic: 2.8.2
pydub: 0.25.1
python-multipart: 0.0.19
pyyaml: 6.0.1
ruff: 0.5.6
safehttpx: 0.1.6
semantic-version: 2.10.0
starlette: 0.41.2
tomlkit: 0.12.0
typer: 0.12.3
typing-extensions: 4.12.2
urllib3: 2.2.2
uvicorn: 0.30.5
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2024.2.0
httpx: 0.27.0
huggingface-hub: 0.25.2
packaging: 24.1
typing-extensions: 4.12.2
websockets: 12.0
```

### Severity

Blocking usage of gradio

**Comment by elismasilva** [comment]
After doing some more tests, I noticed that if I set the height to something like 1000, for example, I can see the buttons. But I think the correct behavior of the component would be to allow me to define the width and height so that it fits in the interface at the size I want, and when uploading the image to the component, even if it is larger than the component's measurements, it is adjusted to fit visibly inside the component, considering the space needed for the buttons below. This resizing would only be visual, not changing the real resolution of the image or the applied mask. I tested ImageMask, ImageEditor, Sketchpad, they all have the same problem.

**Comment by elismasilva** [comment]
Since we're talking about this component, if you could answer me, when I tested this ImageEditor component in version 4.44 it ran extremely slow, I couldn't even draw the mask because it would freeze.
I haven't been able to compare its speed in version 5 yet. But this application here at this link https://huggingface.co/spaces/diffusers/stable-diffusion-xl-inpainting uses an older version of Gradio and runs extremely well, without lag. If you could guarantee that in version 5 it would run like this, it would be perfect.

**Comment by FOSSEnjoyer69** [comment]
I have the same issue, the component does not scale with the size of the imge

**Comment by FOSSEnjoyer69** [comment]
i make a workaround with the code below

`           def set_mask_composite_image(input_image):
    input_image = input_image
    height, width = input_image.shape[:2]

    return gr.update(value=input_image, height=height + int(height * 0.1))
input_image.upload(set_mask_composite_image, inputs=[input_image], outputs=[mask_composite_image], queue=False)
`

**Comment by elismasilva** [comment]
for while i cant implement this component on my interface to user draw the masks, i need upload maskfile created by external tool.

**Comment by elismasilva** [comment]
@pngwn @freddyaboulton this last PR #10357 was suposed to fix this bug too? because ive update to lasted version of gradio and i have same problem.

**Comment by freddyaboulton** [comment]
Yes the latest PR should have fixed the issue. If you set a `height` thats smaller than the canvas size then the canvas will appear cut off. So I would not set `height` in your case.

**Comment by elismasilva** [comment]
@freddyaboulton  see this code:

```python
import gradio as gr

with gr.Blocks() as app:   
    with gr.Row():
        with gr.Column(min_width=400):
            imageMask = gr.ImageMask(
                    type="pil",
                    label="Input Image",                    
                    canvas_size=(400, 280),
                    crop_size=(400, 280),
                    layers=False,
                    sources=["upload"],
                    interactive=True,
                    width=400,
                    height=400
                )

app.launch(inbrowser=True)
```

I need the size of the component on the page to be 400x400px. So I had to decrease the height on the canvas and crop so the brush buttons would appear. But when I load an image into it the image overlaps the buttons and I can't draw the mask on it. I

![Image](https://github.com/user-attachments/assets/fb997bf9-1366-451b-8abc-b1d31d494d51)

**Comment by freddyaboulton** [comment]
Hi @elismasilva! Please set `fixed_canvas=True` if you want the canvas size to be fixed and not resize based on the input image!

To recap the new behavior:
* Image editor always resizes the canvas to match the size of the uploaded image
* `canvas_size` is just the initial canvas size
* Setting a `height` is the total height of the component. If this is smaller than the height of an uploaded image, the component may not fit everything.

**Comment by elismasilva** [comment]
> Hi [@elismasilva](https://github.com/elismasilva)! Please set `fixed_canvas=True` if you want the canvas size to be fixed and not resize based on the input image!
> 
> To recap the new behavior:
> 
> * Image editor always resizes the canvas to match the size of the uploaded image
> * `canvas_size` is just the initial canvas size
> * Setting a `height` is the total height of the component. If this is smaller than the height of an uploaded image, the component may not fit everything.

Thanks, it worked at first. I'm going to implement the component in my interface, if there's any problem I'll get back to you.

**Comment by freddyaboulton** [comment]
I'm glad @elismasilva !

**Comment by elismasilva** [comment]
@freddyaboulton, I again. When fixed_canvas=True, it solved the problem regarding the image display in the component. However, when I pass the component to the event to retrieve the mask image and the original image, they were resized to the values ​​I defined 400x330, but my image is 768x768. When fixed_canvas=False, the image remains in its original resolution when retrieved in the function. Would it be possible to maintain the original resolution even when fixed_canvas=True? In this case, fixed_canvas would only have a display effect. It also seems that cropping is not working, because with values ​​defined for crop_size or without defining it, the retrieved image does not have its size changed, and it is also not affected by the fixed_canvas parameter.

And how would cropping work? If, for example, I have an image of 768x768 and I crop 100 pixels on each side, it would be 668x668. But if it is being displayed at 400x330, in the preview it will be 200x230, but it should return 668x668 when retrieving the event. So I imagine that would be the correct way.

**Comment by pngwn** [comment]
This issue of requiring a specific display size (but a different image resolution) will be addressed in a release shortly (next week or so). The canvas_size should only be used to dictate the image resolution. If fixed_canvas is set then the image will always match this resolution regardless of what image is used. 

The crop_size parameter is now deprecated (this was previously used  to coerce the canvas size). Cropping is a now a UI action only and does not affect the canvas_size. 

Out-painting use cases are not well support currently (cropping is the only way to create a border) but will be better supported shortly when we add image resizing, canvas resizing, and default out-painting borders (that can be set by the developer).

**Comment by elismasilva** [comment]
I see. So, currently if I use fixed_canvas = False with a 1024x1024 image, the canvas size pushes the buttons out of the visible area defined in the component size. I need to have the same result as when I set fixed_canvas = True with the buttons visible but without changing the final resolution of the image. It seems to me that the problem only occurs with square resolution images (768x768, 1024x1024). If I have a 2940x1960 image, rectangular with fixed_canvas=False the buttons appear normally. I don't want to increase the size of the component on the screen because I have limited space in my interface. So I'll wait for you to finish the adjustments to use the component.

**1024x1024**
![Image](https://github.com/user-attachments/assets/e1dc7592-1b1a-4e20-9b4e-b604900198a3)

**2940x1960**
![Image](https://github.com/user-attachments/assets/b88c9701-87b1-4ebe-81cc-a33b8c81ad89)

About cropping. So from what you said, currently it is cropping creating a transparent border in the image, but from what I saw in the final image it is not removing the transparent area. For cropping purposes, I think this would not be correct. I think you can keep the cropping function but it should remove this transparent area from the final image.

**Currently**
![Image](https://github.com/user-attachments/assets/c7ea0b3e-ce61-4e6f-8d7e-1369198a53e4)

**Expected**
![Image](https://github.com/user-attachments/assets/43d8c6cc-377f-4aec-a746-52137e9697d1)

I also found a small bug in this cropping function too: If you cropping one image then delete it and upload other image this happen: 
![Image](https://github.com/user-attachments/assets/f307dc2e-a390-4fe6-b15a-9e15faad3e1d)

For Outpaint cases, you should allow the area that the cropping area selection frame is expandable outwards, re-centering the image but with transparent edges.

I made a sketch in Paint of what it would be like:
![Image](https://github.com/user-attachments/assets/c2c12695-fe81-4b9c-9ca3-42eb1ffdf819)

**Comment by pngwn** [comment]
This was fixed in #10635.

The other issues mentioned should be reproduced with latest and a new issue filed if one does not already exist.
